scripts/prepare_commit.sh

```diff
@@ -57,7 +57,32 @@ if [ -z "$MODIFIED" ]; then
   exit 0
 fi
 
-if [[ -n "$QGIS_CHECK_SPELLING" && -x "${TOPLEVEL}"/scripts/spell_check/check_spelling.sh ]]; then "${TOPLEVEL}"/scripts/spell_check/check_spelling.sh "$MODIFIED"; fi
+if [[ -x "${TOPLEVEL}"/scripts/spell_check/check_spelling.sh ]]; then
+  while :; do # one-iteration loop to early break if reqs are not met
+    which ag > /dev/null || {
+      echo "WARNING: the ag(1) executable was not found, spell checker could not run" >&2
+      break
+    }
+    which unbuffer > /dev/null || {
+      echo "WARNING: the unbuffer(1) executable was not found, spell checker could not run" >&2
+      break
+    }
+    "${TOPLEVEL}"/scripts/spell_check/check_spelling.sh "$MODIFIED"
+    break
+  done
+fi
+
+if [[ -x "${TOPLEVEL}"/tests/code_layout/test_shellcheck.sh ]]; then
+  while :; do # one-iteration loop to early break if reqs are not met
```

Comment:
Wouldn't something like

```sh
if command -v shellcheck >/dev/null 2>&1
then
    ${TOPLEVEL}/tests/code_layout/test_shellcheck.sh || exit 1
else
    echo "WARNING: the shellcheck(1) executable was not found, shell checker could not run" >&2
fi
```

be easier to follow?

Reply:
force pushed to do this